src/NServiceBus.Gateway/Sending/GatewaySendBehavior.cs

```diff
@@ -6,42 +6,45 @@ namespace NServiceBus.Gateway.Sending
     using System.Threading.Tasks;
     using Extensibility;
     using Notifications;
+    using NServiceBus.Gateway.Routing.Sites;
     using NServiceBus.Pipeline;
     using NServiceBus.Routing;
     using Routing;
     using Receiving;
     using Settings;
     using Transports;
-    using IRouteMessagesToSites = NServiceBus.Gateway.Routing.Sites.IRouteMessagesToSites;
 
     class GatewaySendBehavior : PipelineTerminator<ISatelliteProcessingContext>
     {
         readonly IManageReceiveChannels channelManager;
         readonly IDispatchMessages dispatcher;
         readonly ReadOnlySettings settings;
         readonly SingleCallChannelForwarder forwarder;
-        readonly IEnumerable<IRouteMessagesToSites> routers;
+        readonly ConfigurationBasedSiteRouter configRouter;
         readonly MessageNotifier messageNotifier;
         readonly string inputAddress;
```

Comment:
The fields should be at the bottom of the file but I think its better to run a code cleanup as a separate pull after we get this one in so lets leave this as is for now?